specification/app/resource-manager/Microsoft.App/preview/2023-08-01-preview/CommonDefinitions.json

```diff
@@ -0,0 +1,1729 @@
+          "description": "Collection of resources.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/DaprSubscription"
+          }
+        },
+        "nextLink": {
+          "description": "Link to next page of resources.",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "DaprSubscription": {
+      "description": "Dapr PubSub Event Subscription.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ProxyResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "description": "Dapr PubSub Event Subscription resource specific properties",
+          "type": "object",
+          "properties": {
+            "pubsubName": {
+              "description": "Dapr PubSub component name",
+              "type": "string"
+            },
+            "topic": {
+              "description": "Topic name",
+              "type": "string"
+            },
+            "deadLetterTopic": {
+              "description": "Deadletter topic name",
+              "type": "string"
+            },
+            "routes": {
+              "description": "Subscription routes",
+              "$ref": "#/definitions/DaprSubscriptionRoutes"
+            },
+            "scopes": {
+              "description": "Application scopes to restrict the subscription to specific apps.",
+              "type": "array",
+              "items": {
+                "type": "string"
+              }
+            },
+            "metadata": {
+              "description": "Subscription metadata",
+              "type": "object",
+              "additionalProperties": {
+                "type": "string"
+              }
+            },
+            "bulkSubscribe": {
+              "description": "Bulk subscription options",
+              "$ref": "#/definitions/DaprSubscriptionBulkSubscribeOptions"
+            }
+          },
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "DaprSubscriptionBulkSubscribeOptions": {
+      "description": "Dapr PubSub Bulk Subscription Options.",
+      "type": "object",
+      "properties": {
+        "enabled": {
+          "description": "Enable bulk subscription",
+          "type": "boolean",
+          "default": false
+        },
+        "maxMessagesCount": {
+          "description": "Maximum number of messages to deliver in a bulk message.",
+          "type": "integer",
+          "format": "int32"
+        },
+        "maxAwaitDurationMs": {
+          "description": "Maximum duration in milliseconds to wait before a bulk message is sent to the app.",
+          "type": "integer",
+          "format": "int32"
+        }
+      }
+    },
+    "DaprSubscriptionRoutes": {
+      "description": "Dapr PubSub Event Subscription Routes configuration.",
+      "type": "object",
+      "properties": {
+        "rules": {
```

Comment:
Consider adding x-ms-identifiers

refs: https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-identifiers

Reply:
@Juliehzl This property is "routes" or "properties.routes" - not sure what syntax you need here.

See: https://github.com/Azure/azure-rest-api-specs/pull/25631/files#diff-81e032bccc1eeac06f139f4ca0b6b230da77602c93aefb4158a85aa9cd0ec0d4R906